rskj-core/src/test/java/org/ethereum/rpc/Web3ImplTest.java

```diff
@@ -295,6 +296,29 @@ public void sendRawTransaction() throws Exception {
         org.junit.Assert.assertEquals(BigInteger.valueOf(1000000), new BigInteger(1, eth.tx.getValue()));
     }
 
+    @Test
+    public void sendTooMuchGasRawTransaction() throws Exception {
+        Web3Impl web3 = createWeb3();
+        SimpleEthereum eth = new SimpleEthereum();
+        web3.eth = eth;
+
+        Account acc1 = new AccountBuilder().name("acc1").build();
+        Account acc2 = new AccountBuilder().name("acc2").build();
+        Transaction tx = new TransactionBuilder().sender(acc1).receiver(acc2)
+                .value(BigInteger.valueOf(1000000))
+                .gasLimit(new BigDecimal(Math.pow(2, 60)).add(new BigDecimal(1)).toBigInteger()).build();
+
+        String rawData = Hex.toHexString(tx.getEncoded());
+
+        try {
+            web3.eth_sendRawTransaction(rawData);
+        } catch (JsonRpcInvalidParamException e) {
+                Assert.assertEquals(e.getMessage(), "Gas limit exceeds Transaction max value tolerated");
+        } catch (Exception e) {
+            Assert.fail();
```

Comment:
In the above code, the test could pass WITHOUT raising exception. My suggestion: only one catch, AND a return at the end of the catch. THEN AFTER closing the try, put an Assert.fail(). In this way, it is clear that if the catch is not executed, then test FAILS

```
try {
    web3.eth_sendRawTransaction(rawData);
 } catch (JsonRpcInvalidParamException e) {
    Assert.assertEquals(e.getMessage(), "Gas limit exceeds Transaction max value tolerated");
	return;
 }

Assert.fail(); 
```

Reply:
Yes it should check that always an exception is thrown.